src/ITILValidationTemplate.php

```diff
@@ -0,0 +1,283 @@
+        $options['display_emptychoice'] = true;
+        $options['display']             = false;
+        $options['width']               = '95%';
+        $options['multiple']            = true;
+
+        if (empty($name)) {
+            $name = "itemtypes";
+        }
+
+        return Dropdown::showFromArray($name, self::getUseTemplateForValues(), $options);
+    }
+
+    /**
+     * Display specific "approver" field
+     *
+     * @param $value
+     * @param $name
+     * @param $options
+     */
+    public static function displayValidatorField(
+        $value = null,
+        $options = []
+    ) {
+        if ($value) {
+            $itemtype = array_key_first($value);
+            $items_id = array_map(function ($item) {
+                return $item['items_id'];
+            }, $value[$itemtype]);
+        }
+
+        $options['users_id_requester'] = \Session::getLoginUserID();
+        $options['itemtype_target']    = $itemtype ?? null;
+        $options['groups_id']          = $value ? $value[$itemtype][0]['groups_id'] : null;
+        $options['items_id_target']    = $value && $itemtype == 'Group' && count($items_id) == 1 ?
+            $items_id[0] : $items_id ?? null;
+        $options['right']              = 'validate_request';
+        $options['display']            = false;
+
+        return CommonITILValidation::dropdownValidator($options);
+    }
+
+    /**
+     * Get possibles values for "itemtypes" field
+     * @return array timestamp before each bump => label
+     */
+    public static function getUseTemplateForValues(): array
+    {
+        return [
+            'TicketValidation' => __('Ticket Validation'),
+            'ChangeValidation' => __('Change Validation'),
+        ];
+    }
+
+    public function displaySpecificTypeField($id, $field = [], array $options = [])
+    {
+        if ($field['name'] == 'itemtypes') {
+            echo self::displayUseTemplateForfield($this->fields['itemtypes'], "", []);
+        } elseif ($field['name'] == 'approver') {
+            echo self::displayValidatorField(ITILValidationTemplate_Target::getTargets($id), []);
+        }
+    }
+
+    public static function getSpecificValueToDisplay($field, $values, array $options = [])
+    {
+        if ($field == 'itemtypes') {
+            $value = '';
+            foreach (explode(',', $values[$field]) as $key => $val) {
+                $value .= ($key > 0 ? ', ' : '') . self::getUseTemplateForValues()[$val];
+            }
+
+            return $value;
+        }
+
+        return parent::getSpecificValueToDisplay($field, $values, $options);
+    }
+
+    public static function getSpecificValueToSelect($field, $name = '', $values = '', array $options = [])
+    {
+        if ($field == 'itemtypes') {
+            return self::displayUseTemplateForfield($values[$field], $name, $options, false);
+        } elseif ($field == 'approver') {
+            return self::displayValidatorField($values[$field], $name, $options, false);
```

Comment:
These functions only have respectively 3 and 2 arguments but are called with 4 arguments here.